applications/frontend/src/store/reducers/models.ts

```diff
@@ -0,0 +1,30 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { CallOSBApiAction } from '../../middleware/osbbackend'
```

Comment:
Shouldn't have a dependency on the middleware within the reducer. Reducers should be pure functional. So shouldn't call actions and shouldn't interact with other reducers or middlewares.

Reply:
agree with it and solved in my last commit